src/components/hub/ChatList.js

```diff
@@ -45,27 +45,30 @@ export default function ChatList() {
       return <p className="chat-list__empty">No chats are available.</p>;
     }
 
-    return filteredConversations.map((obj) => (
-      <ConversationItem
-        key={obj._id}
-        isSelected={activeConv === obj._id}
-        onClickFunc={() => {
-          dispatch(setSelectedConversation({ id: obj._id }));
-          navigateTo(`/#${obj._id}`);
-        }}
-        chatName={
-          obj.name ||
-          getUserFullName(
-            participants[
-              obj[obj.owner_id === currentUser._id ? "opponent_id" : "owner_id"]
-            ] || {}
-          )
-        }
-        chatObject={obj}
-        currentUserId={currentUser._id}
-      />
-    ));
-  }, [conversations, participants, activeConv, currentUser]);
+    return filteredConversations.map((obj) => {
+      const isSelected = activeConversationId === obj._id;
+      const chatParticipantId =
+        obj.owner_id === currentUser._id ? obj.opponent_id : obj.owner_id;
+      const chatParticipant = participants[chatParticipantId] || {};
+      const chatName = obj.name || getUserFullName(chatParticipant);
+
+      const onClickFunc = () => {
```

Comment:
I'm wondering if there is much sense in creating a separate function for each conversation item. 
Maybe it worth trying to create a single one which will accept an id as parameter